catalog/core/catalog-core-versioning/versioning-common/src/test/groovy/ddf/catalog/core/versioning/HistoryMetacardImplSpecTest.groovy

```diff
@@ -72,18 +75,76 @@ class HistoryMetacardImplSpecTest extends Specification {
-                meta.metacard,
+                meta.metacard as Metacard,
                 action,
                 SecurityUtils.subject)
 
         when:
-        Metacard metacard = history.toBasicMetacard()
+        Metacard metacard = history.getMetacard([BasicTypes.BASIC_METACARD])
+
+        then:
+        metacard != null
+        history.versionOfId == metacard.id
+        metacard.tags.containsAll(meta.tags)
+        !metacard.tags.contains(MetacardVersion.VERSION_TAG)
+        metacard.title == meta.metacard.title
+    }
+
+    def "Non default metacard type back to type from existing list"() {
+        setup:
+        def meta = nonBasicMetacard()
+        Action action = Action.CREATED
+
+        when: "History items are created from non basic metacard types"
+        MetacardVersion history = new MetacardVersion(
+                meta.metacard as Metacard,
+                action,
+                SecurityUtils.subject,
+                [BasicTypes.BASIC_METACARD, meta.metacardType])
+
+        then: "The metacard type should contain non default attribute descriptors"
+        meta.attributeDescriptor in history.metacardType.attributeDescriptors
+        history.metacardType.attributeDescriptors.containsAll(
+                meta.metacard.metacardType.attributeDescriptors)
+
+        when:
+        Metacard metacard = history.getMetacard([BasicTypes.BASIC_METACARD, meta.metacardType])
 
         then:
         metacard != null
-        history.versionOfId.equals(metacard.id)
+        history.versionOfId == metacard.id
         metacard.tags.containsAll(meta.tags)
         !metacard.tags.contains(MetacardVersion.VERSION_TAG)
+        metacard.getAttribute(meta.attributeDescriptor.name)?.value == meta.attributeValue
+        meta.metacard.metacardType.attributeDescriptors.containsAll(metacard.metacardType.attributeDescriptors)
+
+
+    }
+
+    def "Non default metacard type back to type from serialized type"() {
+        setup:
+        def meta = nonBasicMetacard()
+        Action action = Action.CREATED
+
+        when: "History items are created from non basic metacard types"
+        MetacardVersion history = new MetacardVersion(
+                meta.metacard as Metacard,
+                action,
+                SecurityUtils.subject)
+
+        then: "ensure the type-binary attribute is there"
+        history.getAttribute(MetacardVersion.VERSION_TYPE_BINARY)?.value != null
+
+        when: "reconstructing metacard with only serialized type available"
+        Metacard metacard = history.getMetacard([BasicTypes.BASIC_METACARD])
+
+        then: "ensure metacard and type are properly restored"
+        metacard.metacardType.name == meta.metacard.metacardType.name
+        def ad1 = metacard.metacardType.attributeDescriptors
+        def ad2 = meta.metacard.metacardType.attributeDescriptors
+        (ad1 - ad1.intersect(ad2)).isEmpty() // Symmetric difference of set should be empty
```

Comment:
This test isn't sufficient proof here. If `ad1` were an empty set to begin with, you'd have an empty set at the end. I think you also want to assert that `ad1` isn't empty to begin with.